src/app/download/download.component.ts

```diff
@@ -1,14 +1,28 @@
 import {Component} from '@angular/core';
 import {DownloadService} from './download.service';
 import * as JSZip from 'jszip';
+import {AdAuthoringService} from '../ad-authoring/ad-authoring.service';
+import {Observable} from 'rxjs';
+import {AdAuthoringWorkflowState} from '../ad-authoring/ad-authoring.state';
 
 @Component({
   selector: 'app-download',
   templateUrl: './download.component.html',
   styleUrls: ['./download.component.scss'],
 })
 export class DownloadComponent {
-  constructor(private downloadService: DownloadService) {}
+  disabledVal = false;
+  adAuthoringObs: Observable<AdAuthoringWorkflowState>;
+
+  constructor(
+    private downloadService: DownloadService,
+    private adAuthoringService: AdAuthoringService
+  ) {
+    this.adAuthoringObs = adAuthoringService.getAdAuthorings();
+    this.adAuthoringObs.subscribe(
```

Comment:
We actually don't need to subscribe to the adAuthoringService here. Instead we can have getDisabledVal return an Observable<boolean> that emits on state changes. Otherwise we'll need to manage the lifecycle of this subscription.

Reply:
How can I return an observable that only emits on state changes?